versions.tf

```
@@ -2,6 +2,6 @@ terraform {
   required_version = ">= 0.12.6, < 0.14"
 
   required_providers {
-    aws = "~> 2.57"
+    aws = "~> 3.0"
```

This is a really bad way to achieve what you're trying to do. Now the module is not compatible with v2 of the provider, which is a breaking change, and requires a major version bump. For this single line change.

The [Terraform documentation](https://www.terraform.io/docs/configuration/modules.html#provider-version-constraints-in-modules) recommends the use of `>=` in `required_providers` for exactly this reason. No release would have been necessary to support the new provider.

A better solution would be:
```suggestion
    aws = ">= 2.57"
```
Would only need a minor version release of the module.

Alternatively, if Anton wants to stay guarded about the AWS provider version:
```hcl
  aws = ">= 2.57, < 4.0"
```

---

You're totally right 🤦